contributing/DOCS-CONTRIBUTING.md

```diff
@@ -68,45 +71,203 @@ documentation efforts, who are happy to help you get started.
+
+    Depending on the Knative component covered by the code sample that you
+    want to contribute, your PR should add that sample in one of the following
+    folders:
+
+    * Build samples: [`/docs/build/samples`](https://github.com/knative/docs/tree/master/docs/build/samples)
+    * Eventing samples: [`/docs/eventing/samples`](https://github.com/knative/docs/tree/master/docs/eventing/samples)
+    * Serving samples: [`/docs/serving/samples`](https://github.com/knative/docs/tree/master/docs/serving/samples)
+
+    Important: If these samples become outdated, receive no traffic, or lack
+    an active maintainer, they will be moved into the "community samples"
+    folder.
+
+  * **Community owned and maintained samples**: Code samples that have
+  been contributed by Knative community members. These samples might not receive
+  regular maintenance. It is possible that a sample is no longer current and
+  is not actively maintained by its original author. While we encourage a
+  contributor to maintain their content, we acknowledge that it's not always
+  possible for certain reasons, for example other commitments and time
+  constraints.
+
+   While a sample might be out of date, it could still provide assistance and
+   help you get up-and-running with certain use-cases. If you find that
+   something is not right, contains out-dated info, please help us become aware
+   of the issue. In which case that sample might be fix if bandwidth or resource
+   exists, or taken down and archived into a past release branch (where it
+   hopefully worked last).
+
+#### Choosing the correct branch
+
+  It is likely that your docs contribution is either for new or changed
+  features in the product, or for a fix or update existing content.
+
+  * **New or changed features**:
+    If you are adding or updating documentation for a new or changed feature,
+    you likely want to open your PR against the `master` branch. All pre-release
+    content for active Knative development belongs in
+    [`master`](https://github.com/knative/docs/tree/master/).
+
+  * **Fixes and updates**:
+    If you find an issue in a past release, for example a typo or out-of-date
+    content, you likely need to open multiple and subsequent PRs. If not a
+    followup PR, at least add the "`cherrypick` labels" to your original PR
+    to indicate in which of the past release that your change affects.
+
+     For example, if you find a typo in a page of the `v0.5` release, then that
+     page in the `master` branch likely also has that typo.
+
+     To fix the typo:
+     1. You open a PR against
+        [`master`](https://github.com/knative/docs/tree/master/).
+     1. Add one or more `cherrypick-#.#` labels to that PR to indicate which
+        of the past release branches should also be fixed.
+     1. If you want to complete the fix yourself (best practice), you then open
+        a subsequent PR by running `git cherry-pick [OriginalPRCommit#]` against
```

Comment:
I don't know if the "Original PR commit" reference here makes sense; I wait until the PR is merged and then grab the squashed commit number that shows at the bottom. For example: https://github.com/knative/docs/pull/1143 shows `3f8fbb2`

Reply:
ive used the language from the template, ptal